test/e2e/storage/volume_metrics.go

```diff
@@ -95,6 +100,53 @@ var _ = SIGDescribe("[Serial] Volume metrics", func() {
 			verifyMetricCount(storageOpMetrics, updatedStorageMetrics, volumeOp)
 		}
 	})
+
+	It("should create prometheus metrics for volume summary stats", func() {
+		var err error
```

Comment:
A lot of the code in this new `it` block is duplicated from the previous `It`. Consider moving some(most) of the dup code into the `BeforeEach`. The common `defer` can be changed to an `AfterEach`.

Reply:
I was thinking about that but there's duplication but some subtle differences in terms of ordering in terms of when initial metrics are collected and PVC/Pod created between the two variations.

In terms of pod deletion and PVC cleanup - that should be able to be moved into a common AfterEach.

Reply:
if you move pod deletion in common `Aftereach` step, `volume_detach` metric will fail because it expects pod deletion to happen within `It` block itself.  We can extract duplicate stuff into a function or something and still keep same structure if that is easier.

Reply:
@gnufied what's the volume_detach difference between a `defer` called immediately before the It() returns vs. an `AfterEach` called right after the It() returns?

Reply:
the pod is not being deleted in a `defer` block, it is being deleted inline. So if you move the pod deletion to `Aftereach` no detach will fire at the time where metrics are being calculuated.

Reply:
@gnufied yes, this makes sense. In my comment about changing the `defer` to an `AfterEach` I did not mean to include pod deletion in the AfterEach.